File: LECTURE-CODES/WEEK-02/gapminder.r

library('tidyverse')

print("----ORIGINAL----")
gap <- read_csv('data/gapminder.csv')
head(gap)

print("----GROUP BY CONTINENT----")
gap %>% group_by(continent)

print("----AVERAGE LIFE EXPECTANCY BY CONTINENT----")
gap %>% group_by(continent) %>% summarize(avgLifeExp = mean(lifeExp))


print("----AVERAGE LIFE EXPECTANCY BY COUNTRY----")
gap %>% group_by(country) %>% summarize(avgLifeExp = mean(lifeExp))

print("----AVERAGE LIFE EXPECTANCY BY CONTINENT AND YEAR----")
gap %>% group_by(continent, year) %>% summarize(mean(lifeExp))

print("----AVERAGE+MEDIAN EXPECTANCY BY YEAR----")
gap %>% group_by(year) %>% summarize(means = mean(lifeExp), medians = median(lifeExp))

File: LECTURE-CODES/WEEK-02/gapminder.py
import numpy  as np
import pandas as pd

df = pd.read_csv('data/gapminder.csv')
print("----ORIGINAL----")
print(df.head())

print("----GROUP BY CONTINENT----")
print((df.groupby('continent')).head())
# print('----\n',df.groupby('continent').tail())

print("----AVERAGE LIFE EXPECTANCY BY CONTINENT----")
print(df.groupby('continent')['lifeExp'].mean())

print("----AVERAGE LIFE EXPECTANCY BY COUNTRY----")
print(df.groupby('country')['lifeExp'].mean())

print("----AVERAGE LIFE EXPECTANCY BY CONTINENT AND YEAR----")
print(df.groupby(['continent','year'])['lifeExp'].mean())

print("----AVERAGE+MEDIAN EXPECTANCY BY YEAR----")
print(df.groupby('year')['lifeExp'].agg([np.mean,np.median]))